python/ray/util/client/server/server.py

```diff
@@ -564,7 +564,10 @@ def unify_and_track_outputs(self, output, client_id):
 def return_exception_in_context(err, context):
     if context is not None:
         context.set_details(encode_exception(err))
-        context.set_code(grpc.StatusCode.INTERNAL)
+        # Note: https://grpc.github.io/grpc/core/md_doc_statuscodes.html
+        # INVALID_ARGUMENT used here since it should never be generated by the
+        # grpc lib -- this way we know the error was generated by ray logic
+        context.set_code(grpc.StatusCode.INVALID_ARGUMENT)
```

Comment:
Would ABORTED be a more appropriate error code here? Conventions in my past team is to use ABORTED to signal application errors that are potentially retriable.